Main

ARMINSHIVES opened the pull request:
See Updated ReadMe for changes

Commit:
modified:   README.md
	new file:   data/config.py
	new file:   data/database/faiss/.gitkeep
	new file:   data/database/faiss/openfoam_allrun_scripts/index.faiss
	new file:   data/database/faiss/openfoam_allrun_scripts/index.pkl
	new file:   data/database/faiss/openfoam_command_help/index.faiss
	new file:   data/database/faiss/openfoam_command_help/index.pkl
	new file:   data/database/faiss/openfoam_tutorials_details/index.faiss
	new file:   data/database/faiss/openfoam_tutorials_details/index.pkl
	new file:   data/database/faiss/openfoam_tutorials_structure/index.faiss
	new file:   data/database/faiss/openfoam_tutorials_structure/index.pkl
	new file:   data/foamgpt/config.py
	new file:   data/foamgpt/data/foamgpt_all.jsonl
	new file:   data/foamgpt/data/foamgpt_openai_test.jsonl
	new file:   data/foamgpt/data/foamgpt_openai_train.jsonl
	new file:   data/foamgpt/data/foamgpt_test.jsonl
	new file:   data/foamgpt/data/foamgpt_train.jsonl
	new file:   data/foamgpt/data/foamgpt_user_requirements.jsonl
	new file:   data/foamgpt/data/parsed_openfoam_cases.jsonl
	modified:   data/foamgpt/foamgpt_gen.py
	modified:   data/foamgpt/foamgpt_huggingface.py
	new file:   data/foamgpt/tracking_aws.py
	new file:   data/foamgpt/utils.py
	new file:   data/raw/openfoam_allrun_scripts.txt
	new file:   data/raw/openfoam_case_stats.json
	new file:   data/raw/openfoam_command_help.txt
	new file:   data/raw/openfoam_commands.txt
	new file:   data/raw/openfoam_tutorials_details.txt
	new file:   data/raw/openfoam_tutorials_structure.txt
	new file:   data/utils.py
	new file:   environment.yml
	new file:   finetune/Results/Eval/eval_loss.png
	new file:   finetune/Results/Eval/eval_mean_token_accuracy.png
	new file:   finetune/Results/Eval/eval_num_tokens.png
	new file:   finetune/Results/Eval/eval_runtime.png
	new file:   finetune/Results/Eval/eval_samples_per_second.png
	new file:   finetune/Results/Eval/eval_steps_per_second.png
	new file:   finetune/Results/Train/train_epoch.png
	new file:   finetune/Results/Train/train_grad_norm.png
	new file:   finetune/Results/Train/train_learning_rate.png
	new file:   finetune/Results/Train/train_loss.png
	new file:   finetune/Results/Train/train_mean_token_accuracy.png
	new file:   finetune/Results/Train/train_num_tokens.png
	new file:   finetune/config.py
	modified:   finetune/finetuning_script.py
	new file:   finetune/utils.py

TingwenZhang reviewed (commented):
Put the packages in an `environment.yml`. In README.md just put the instruction as
```bash
conda env create --file environment.yml
```

TingwenZhang commented on README.md:
````diff
@@ -21,3 +21,148 @@ Please type out the following command into command prompt or other terminals and
   ```bash
   python finetuning_script.py
   ```
+
````
In README.md just put the instruction as
```bash
conda env create --file environment.yml
```

TingwenZhang commented on data/foamgpt/foamgpt_gen.py:
```diff
@@ -181,6 +181,8 @@ def main():
     config = Config()
     print(f"Config: {config}")
+    print("config.model provider", config.model_provider)
+    print("config.model version", config.model_version)
```
These two lines were added for debugging. You can remove them.

TingwenZhang commented on finetune/config.py:
```diff
@@ -0,0 +1,20 @@
+# config.py
```
Why are there three `config.py`? This one is still using OpenAI, should be Claude.

lingyue404 replied:
For the default config for public, using OpenAI is better. OpenAI is more popular and It is easier to configure than bedrock claude API.

TingwenZhang reviewed (commented):
I see three `config.py`: [data/config.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-41b42b15b59a1870aca091026818b08fc2f1505646006151977acd97c45f642f), [foamgpt/config.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-7095381c5b8b0e0d1b79af4caa69b7be9e52d2f765b8a6c9471bc238983ab789), [finetune/results/Train/config.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-d90e937451575e0459e84ec26cac8ea5a78841da306e3f2ec41e3797dbea72d4)

and three `utils.py` [foamgpt/data/utils.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-d7968a0300da23fde22648e7985c118af316b5d39e8a65c3e80826a8d0fe44f7), [foamgpt/raw/utils.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-ddfa045583d04b5b91717cba9582c94db2624fe3652890b1002305cd691cf1c9), [finetune/results/Train/utils.py](https://github.com/csml-rpi/FoamGPT/pull/2/files#diff-ecb5d10bab90db0c2f3ccd8c78de4eca37f778c48ee81beed80f18d88e42ee28)

What are each of file above for? If possible, could you merge to only one config and one utils?

Put `database/faiss/` in a `.gitignore` file.

I also left comments in other files

Commit:
removed unnecessary files

Commit:
modified:   README.md

Commit:
modified:   .gitignore

Commit:
removed untracked files

Commit:
restored .gitkeep; removed extra files

Commit:
modified:   .gitignore

Commit:
restored .gitkeep; removed extra files

lingyue404 commented on data/foamgpt/foamgpt_huggingface.py:
```diff
@@ -10,7 +10,7 @@
 
 # Hugging Face configuration
-repo_id = "LeoYML/FoamGPT"
+repo_id = "ARMINSHIVES/FoamGPT"
```
Please don't change it.

lingyue404 commented on data/foamgpt/tracking_aws.py:
Please don't upload the private LLM service related file.

lingyue404 commented on data/utils.py:
Please remove the unnecessary code. If one function is never used in FoamGPT, please remove it.

lingyue404 commented on environment.yml:
Please make sure all the package is nessary for FoamGPT.

lingyue404 commented on finetune/finetuning_script.py:
Don't change the ID to ARMINSHIVES, please double-check all your code.

lingyue404 commented on finetune/finetuning_script.py:
```diff
@@ -49,7 +49,7 @@ def tokenize_data(example):
     device_map="auto",
     trust_remote_code=True,
-    torch_dtype=torch.bfloat16,
+    torch_dtype=torch.float16,
```
why change this?

lingyue404 commented on finetune/finetuning_script.py:
```diff
@@ -78,17 +78,18 @@ def tokenize_data(example):
-    per_device_train_batch_size=2,
-    per_device_eval_batch_size=2,
-    gradient_accumulation_steps=4, 
+    per_device_train_batch_size=1,
```
Why change these parameters?

Commit:
removed unneccesary functions from util.py, reverted to original hyperparamaters for finetuneing_script.py, changed to public repo LeoYML/FoamGPT

Commit:
modified:   data/foamgpt/foamgpt_gen.py

Commit:
Removed "print_statistics(self)" function from LLMServce since its already defined in the code

Commit:
Added some details in README.md to improve User Interaction

Commit:
modified:   README.md

Commit:
Finished Editing Readme. Reformatted for reading clarity and provided insight on hyperparamaters for memory and configuration alignment

Reverted finetuning_script.py to initial structure

Commit:
modified:   README.md

TingwenZhang commented on data/foamgpt/tracking_aws.py:
Remove this file.

TingwenZhang commented on environment.yml:
Use the shorter version I sent you.

TingwenZhang commented on data/faiss/.gitkeep:
Do
```bash
touch data/faiss/.gitkeep
```
to add the `.gitkeep` back.

TingwenZhang commented on data/raw/.gitkeep:
Do
```bash
touch data/raw/.gitkeep
```
to add the `.gitkeep` back.

TingwenZhang commented on .gitignore:
```diff
@@ -205,3 +205,18 @@ cython_debug/
+data/database/faiss/
+!data/database/faiss/.gitkeep
+
+foamgpt/data
```
Add `!foamgpt/data/.gitkeep` after line 221.

TingwenZhang reviewed (commented):
Double check with the current main to make sure all folders with `.gitkeep` still exist.

Do 
```bash
git fetch origin
git diff origin/main
```
to make sure all changes are desired.

Commit:
Made edits suggested by Tingwen

Commit:
modified:   .gitignore
	new file:   data/database/faiss/.gitkeep
	new file:   data/faiss/.gitkeep
	new file:   data/foamgpt/data/.gitkeep
	new file:   data/raw/.gitkeep

Commit:
modified:   .gitignore
